pgbouncer-mixin/panels.libsonnet

```diff
@@ -0,0 +1,156 @@
+local g = import './g.libsonnet';
+local commonlib = import 'common-lib/common/main.libsonnet';
+local utils = commonlib.utils;
+{
+  new(this):
+    {
+      local t = this.grafana.targets,
+      local stat = g.panel.stat,
+      // create stat panel using commonlib
+      clientsWaitingConnections:
+        commonlib.panels.generic.stat.base.new(
+          'Client waiting connections',
+          targets=[t.clientsWaitingConnections],
+          description='Current number of client connections waiting on a server connection.'
+        )
+        + stat.options.withGraphMode('none')
+        + stat.standardOptions.color.withMode('thresholds')
+        + stat.standardOptions.thresholds.withSteps([
+          stat.thresholdStep.withColor('super-light-green')
+          + stat.thresholdStep.withValue(0),
+          stat.thresholdStep.withColor('super-light-orange')
+          + stat.thresholdStep.withValue(10),
+          stat.thresholdStep.withColor('super-light-red')
+          + stat.thresholdStep.withValue(20),
+        ]),
+      activeClientConnections:
+        commonlib.panels.generic.stat.info.new(
+          'Active client connections',
+          targets=[t.activeClientConnections],
+          description='Current number of active client connections.'
+        )
+        + stat.options.withGraphMode('none'),
+      activeServerConnections:
+        commonlib.panels.generic.stat.info.new(
+          'Active server connections',
+          targets=[t.activeServerConnections],
+          description='Current number of client connections that are linked to server connection and able to process queries.'
+        )
+        + stat.options.withGraphMode('none'),
+      maxDatabaseConnections:
+        commonlib.panels.generic.stat.info.new(
+          'Max database connections',
+          targets=[t.maxDatabaseConnections],
+          description='Maximum number of allowed connections for database.'
+        )
+        + stat.options.withGraphMode('none'),
+      maxUserConnections:
+        commonlib.panels.generic.stat.info.new(
+          'Max user connections',
+          targets=[t.maxUserConnections],
+          description='Maximum number of server connections per user allowed.'
+        )
+        + stat.options.withGraphMode('none'),
+      maxClientConnections:
+        commonlib.panels.generic.stat.info.new(
+          'Max client connections',
+          targets=[t.maxClientConnections],
+          description='Maximum number of client connections allowed.'
+        )
+        + stat.options.withGraphMode('none'),
+
+      queriesPooled:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Queries processed',
+          targets=[t.queriesPooled],
+          description=|||
+            Rate of SQL queries pooled by PgBouncer.
+          |||
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('queries/s'),
```

Comment:
I don't remember. is it "safe" to do this? (would the graph ever switch to a scale that wasn't "per second"? would it be better to use a built in unit like ops per second?